### ISSUE TITLE
Can't remove modification flag of shared link within Android App

### ISSUE THREAD

**Issue by CarlosCastaneda** [nextcloud/android#6987]
### Steps to reproduce
1. Create shared link for a folder 
2. Add modificatiin flag to link
3. Removing flag dies mit Work (has to be done on Web or iOS)

### Expected behaviour
- Tell us what should happen
Option flag should bei removed

### Actual behaviour
- Tell us what happens
Option flag won't bei removed even If selected an displayed empty

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version: 7.1.1

Device model:  Moto E4

Stock or customized system: Stock

Nextcloud app version: 3.13.1

Nextcloud server version: 19.0.3.1

Reverse proxy:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Happens also on master

**Comment by tobiasKaminsky** [comment]
This is fixed via #6504 and will be in next 3.14.
